python/lib/modeldata/dmod/modeldata/data/item_domain_detector.py

```diff
@@ -75,8 +81,11 @@ def detect(self, **kwargs) -> DataDomain:
         cat_restriction = self._get_cat_restriction()
         data = StringIO(self._item.decode(self._decode_format)) if isinstance(self._item, bytes) else self._item
```

Comment:
We might want to leave a TODO here so create a wrapper type that takes `bytes` and returns those types via a `read()` method. So, something like:

```python
from dataclasses import dataclass

@dataclass
class BytesReader:
  buff: bytes

  def read(self, size: int = -1) -> bytes:
    if size < 0:
      size = len(self.buff)
    ret = self.buff[:size]
    self.buff = self.buff[len(ret):]
    return ret
```

We could easily do the same thing for `str`'s too. These are nice convenience wrappers around the `Reader` interface that avoid having to go through something like `StringIO` since `pd.read_csv` accepts a `Reader` as an input.

Reply:
As suggested in #609 by @christophertubbs, I did go ahead and modify this to use a _BytesIO_ object rather than a _StringIO_, but we can still address the original question for this comment thread in #609.